unit-tests/live/syncer/test-throughput.cpp

```diff
@@ -0,0 +1,214 @@
+    typedef enum configuration
+    {
+        IR_ONLY,
+        IR_RGB_EXPOSURE,
+        STOP
+    }configuration;
+
+    class ir_cfg
+    {
+    public:
+        ir_cfg(sensor rgb_sensor,
+            sensor ir_sensor,
+            std::vector<rs2::stream_profile > rgb_stream_profile,
+            std::vector<rs2::stream_profile > ir_stream_profile) : _rgb_sensor(rgb_sensor), _ir_sensor(ir_sensor),
+            _rgb_stream_profile(rgb_stream_profile), _ir_stream_profile(ir_stream_profile) {}
+         
+        void test_configuration(configuration test)
+        {
+            std::cout << "==============================================" << std::endl;
+            std::string cfg = test == IR_ONLY ? "IR Only" : "IR + RGB";
+            std::cout << "Configuration: " << cfg << std::endl << std::endl;
+            
+            start_streaming(test);
+            bool exposure_cfg[2] = { false, true }; // true means setting exposure to 180000 and false means setting exposure to 1 (min value)
+            for (auto exposure : exposure_cfg)
+            {
+                if (!set_ir_exposure(exposure))
+                {
+                    std::cout << "ERROR : Setting Exposure value had failed!" << std::endl;
+                    std::cout << "Configuration is stopped" << std::endl;
+                    return;
+                }
+                process_validate_frames(test);
+            }
+            stop_streaming(test);
+        }
+
+    private:
+        bool set_ir_exposure(bool exposure)
+        {
+            std::lock_guard<std::mutex> lock(_mutex); // prevent processing frames while setting exposure
+            _frames_arrival_info.clear();
+            if (_ir_sensor.supports(RS2_OPTION_EXPOSURE))
+            {
+                if (exposure)
+                    REQUIRE_NOTHROW(_ir_sensor.set_option(RS2_OPTION_EXPOSURE, 18000)); // set exposure value to x > 1000/fps
+                else
+                    REQUIRE_NOTHROW(_ir_sensor.set_option(RS2_OPTION_EXPOSURE, 1)); // set exposure value to min value
+                std::this_thread::sleep_for(std::chrono::milliseconds(200)); // wait 200 msec to process FW command
+                _exposure = _ir_sensor.get_option(RS2_OPTION_EXPOSURE);
+                return true;
+            }
+            return false;
+        }
+        void process_validate_frames(configuration test)
+        {
+            _prev_fps = 0.0f;
+            auto process_frame = [&](const rs2::frame& f)
+            {
+                std::lock_guard<std::mutex> lock(_mutex);
+                auto stream_type = std::string(f.get_profile().stream_name());
+                // Only IR fps is relevant for this test, IR1 and IR2 have same fps so it is enough to get only one of them
+                if (stream_type != "Infrared 1")
+                    return;
+                auto frame_num = f.get_frame_number();
+                _actual_fps = (float)f.get_frame_metadata(RS2_FRAME_METADATA_ACTUAL_FPS);
```

Comment:
The call may throw. Check first with `<>.supports()` before dereferencing the data